pilot/pkg/serviceregistry/kube/controller/namespacecontroller.go

```diff
@@ -56,59 +50,42 @@ type NamespaceController struct {
 	getData func() map[string]string
 	client  corev1.CoreV1Interface
 
-	queue queue.Instance
-
-	// Controller and store for namespace objects
-	namespaceController cache.Controller
-	// Controller and store for ConfigMap objects
-	configMapController cache.Controller
+	queue              queue.Instance
+	namespacesInformer cache.SharedInformer
+	configMapInformer  cache.SharedInformer
 }
 
 // NewNamespaceController returns a pointer to a newly constructed NamespaceController instance.
-func NewNamespaceController(data func() map[string]string, options Options, kubeClient kubernetes.Interface) *NamespaceController {
+func NewNamespaceController(data func() map[string]string, kubeClient kube.Client) *NamespaceController {
 	c := &NamespaceController{
 		getData: data,
 		client:  kubeClient.CoreV1(),
 		queue:   queue.NewQueue(time.Second),
 	}
 
-	watchedNamespaceList := strings.Split(options.WatchedNamespaces, ",")
```

Comment:
Is multi-namespaces not supported though this is not perfect now?

Reply:
There is work to add this back correctly, see https://github.com/istio/istio/issues/24640#issuecomment-654309550

Reply:
which will be done before 1.7